sdk/src/main/java/com/akkaserverless/javasdk/eventsourcedentity/EventSourcedEntityProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright 2021 Lightbend Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.akkaserverless.javasdk.eventsourcedentity;
+
+import com.akkaserverless.javasdk.impl.eventsourcedentity.EventSourcedEntityHandler;
+import com.google.protobuf.Descriptors;
+
+public interface EventSourcedEntityProvider<S, E extends EventSourcedEntityBase<S>> {
```

Comment:
Should we add some "Not for user extension" here or do we consider it actual API for user extension as well as common (internal) interface for generated code?

Reply:
I'll add a comment that the concrete implementation is generated, to start with.

Reply:
Good question. 

Let's be restrictive and eventually relax the rules if needed. We have all the interest in keep it 'private'.

Reply:
we can't make it completely internal because it's the parameter to `register`

Reply:
what I mean by 'private' is that it's not real private, but marked as "Not for user extension"